app/parse.py

```diff
@@ -15,8 +24,125 @@ class Product:
+    title = product_element.find_element(
+        By.CLASS_NAME, "title"
+    ).get_attribute("title")
+
+    price = float(product_element.find_element(
+        By.CLASS_NAME, "pull-right"
+    ).text.replace("$", ""))
+
+    description = product_element.find_element(
+        By.CLASS_NAME, "description"
+    ).text
+
+    rating_element = product_element.find_element(By.CLASS_NAME, "ratings")
+    views = rating_element.find_elements(By.CSS_SELECTOR, "p")
+    num_of_reviews = int(views[0].text.split()[0])
+    rating = len(views[-1].find_elements(By.CSS_SELECTOR, "span"))
+
+    return Product(
+        title=title,
+        description=description,
+        price=price,
+        rating=rating,
+        num_of_reviews=num_of_reviews
+    )
+
+
+def save_to_csv(file_name: str, products: list[Product]) -> None:
+    with open(file_name, "w", newline="") as csvfile:
+        fieldnames = [
+            "title",
+            "description",
+            "price",
+            "rating",
+            "num_of_reviews"
+        ]
+
+        csvwriter = csv.writer(csvfile)
+        csvwriter.writerow(fieldnames)
+
+        for product in products:
+            csvwriter.writerow([
+                product.title,
+                product.description,
+                product.price,
+                product.rating,
+                product.num_of_reviews
+            ])
+
+
+def close_cookies(driver: WebDriver) -> None:
+    if len(driver.find_elements(By.CLASS_NAME, "acceptCookies")) > 0:
+        button = driver.find_element(By.CLASS_NAME, "acceptCookies")
+        if button.is_displayed():
+            ActionChains(driver).click(button).perform()
+
+
+def load_all_content(driver: WebDriver) -> None:
+    if len(driver.find_elements(
+            By.CLASS_NAME, "ecomerce-items-scroll-more"
+    )) > 0:
+        button = driver.find_element(
+            By.CLASS_NAME, "ecomerce-items-scroll-more"
+        )
+        if button.is_displayed():
+            ActionChains(driver).click(button).perform()
+            load_all_content(driver)
+
+
+def scrape_page(name: str, url: str) -> None:
+    with ChromeDriver() as driver:
+        driver.get(url)
+
+        close_cookies(driver)
+        load_all_content(driver)
+
+        products_card = driver.find_elements(By.CLASS_NAME, "thumbnail")
+
+        products = [extract_product(product) for product in products_card]
+
+        save_to_csv(f"{name}.csv", products)
+
+
 def get_all_products() -> None:
-    pass
+    pages = {
+        "home": "",
+        "computers": "computers",
+        "laptops": "computers/laptops",
+        "tablets": "computers/tablets",
+        "phones": "phones",
+        "touch": "phones/touch",
+    }
+
+    with ThreadPoolExecutor(max_workers=len(pages)) as executor:
+        futures = [
+            executor.submit(scrape_page, name, f"{HOME_URL}{url}")
+            for name, url in pages.items()
+        ]
+
+    for future in futures:
+        future.result()
```

Comment:
If you meant to execute these futures concurrently, this will not work. This will just call the future one by one with no performance boost